omod/src/main/java/org/openmrs/module/appointmentscheduling/rest/resource/openmrs1_9/AppointmentResource1_9.java

```diff
@@ -197,6 +197,16 @@ protected PageableResult doSearch(RequestContext context) {
 		Visit visit = context.getParameter("visit") != null ? Context.getVisitService().getVisitByUuid(
 				context.getParameter("visit")) : null;
 
+		if (statuses != null && (statuses.get(0).equals(AppointmentStatus.EARLY))) {
+			return new NeedsPaging<Appointment>(Context.getService(AppointmentService.class).getEarlyAppointments(
+					fromDate, toDate, location, provider, appointmentType, statuses.get(0), visitType), context);
+		}
+
+		if (statuses != null && (statuses.get(0).equals(AppointmentStatus.LATE))) {
+			return new NeedsPaging<Appointment>(Context.getService(AppointmentService.class).getLateAppointments(
+					fromDate, toDate, location, provider, appointmentType, statuses.get(0), visitType), context);
+		}
+

```

Comment:
So the design problem I'm seeing here is that you are using the new EARLY and LATE statuses solely as a query parameter.   That is, the underlying appointments in the database will never end up with statuses of EARLY and LATE.

Instead of modifying the existing AppointmentResource, I would create a custom search handler for getting early and late appointments that uses the underlying getEarlyAppointments and getLateAppoinments methods that you have added.

For an example of a custom search handler, see:

https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/PatientByIdentifierSearchHandler1_8.java

https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_9/PatientByIdentifierSearchHandlerTest1_9.java

Reply:
I'll follow the links and see try implement the required funtionality in a similar way

Reply:
Cool, thanks!  You don't necessarily need to go the search handler route, but I was trying to think of a way around what I thing the design issue is:  you don't want (or need) to create EARLY and LATE statuses, as you are calculating those on the fly in the getEarlyAppointments and getLateAppointments methods... you really just need new endpoints to expose "getEarlyAppointments" and "getLateAppointments".